[Infra] Simplify inventory model chart objects

crespocarlos opened the pull request:
part of https://github.com/elastic/kibana/issues/175446

## Summary

Prior refactors on the chart objects did not make their usage flexible enough. Previously, the inventory model held dashboard configs, for https://github.com/elastic/kibana/issues/175446 we need to be able to select charts individually to build the different sections in the UI

The UI components will now be responsible for grouping the necessary charts 


### For reviewers

- Most of the changes are in the `metrics-data-access` plugin, and about removing the `dashboard` folder and improving the chart static config in `charts` foder. They no longer receive any dynamic value.
- in `infra` plugin, places that are displaying charts (kpi, asset details, hosts view) are now grouping the charts manually, instead of using an object from `metrics-data-access` that contained the full dashboard config.
- Host KPIs component was centralized to be reused both in the Hosts View and asset details.e
- Asset Details embeddable component was removed - It was not being used.

crespocarlos committed:
Simplify inventory model chart objects

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `/oblt-deploy-serverless` : Deploy a serverless Kibana instance using the Observability test environments.
- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)

</p>
</details>

crespocarlos committed:
Centralize host KPIs in a single component

Translations and files renaming

Remove unused file

Fix translations

Fix types and improve readability

Fix KPIs decimal place

crespocarlos commented:
/ci

crespocarlos commented:
/ci

crespocarlos commented:
/ci

crespocarlos commented:
/ci

crespocarlos commented:
/ci

crespocarlos commented:
/ci

crespocarlos committed:
Fix test

crespocarlos commented:
/ci

crespocarlos committed:
Clean up

crespocarlos committed:
Revert "Clean up"

This reverts commit db10a19c43f0161dfa24196a1a8b231be5f49bb7.

crespocarlos committed:
Clean up

crespocarlos committed:
Merge branch 'main' of github.com:elastic/kibana into 175446-refactor-chart-static-objects

crespocarlos commented:
/ci

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

crespocarlos commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into 175446-refactor-chart-static-objects

jennypavlova commented on x-pack/plugins/observability_solution/infra/public/pages/metrics/hosts/components/kpis/kpi.tsx:
```diff
@@ -4,22 +4,21 @@
-  const { isRequestRunning: hostCountLoading } = useHostCountContext();
-  const { dataView } = useDataView({ index: (chartProps.dataset as LensDataviewDataset)?.index });
+  const { isRequestRunning: hostCountLoading, data: hostCountData } = useHostCountContext();
+  const { dataView } = useMetricsDataViewContext();
```
I tried to test locally with oblt cluster data and I see infinite loading for the Logs chart:

https://github.com/elastic/kibana/assets/14139027/e532e7a0-f844-4c4c-888c-099f00eb3558


I checked the changes in this PR several times I am not sure if the issue comes from it but I think maybe because the logs chart is using the logs dataview it could be around that change. In general, it should be an error in case my env is misconfigured or the logs are not available there (for example) - so it could be also something around the error handling. I checked on oblt cluster and the same host looks fine there:
<img width="1920" alt="image" src="https://github.com/elastic/kibana/assets/14139027/5c09dbd2-d256-4939-a163-341db186e143">

crespocarlos replied:
I couldn't reproduce the problem. 

<img width="805" alt="image" src="https://github.com/elastic/kibana/assets/2767137/f05e9290-0cb6-4dc2-9897-c97cae12d544">

When there is an error in the logs indices config, this is usually what happens:

<img width="874" alt="image" src="https://github.com/elastic/kibana/assets/2767137/7b9b9b1c-0e96-45ff-9330-46f416d2c0d9">

I'm confused whether you think the endless loading problem is caused by the `kpi` component

jennypavlova reviewed (approved):
Well done 💯  Thanks for doing the refactoring! 
~~I found I small issue with the logs chart and added a comment, can you please take a look? ( I am not sure that the PR causes the issue so I will approve the PR as the code looks good 👍 )~~ - Not related
